task-annotation-generator.js

```diff
@@ -0,0 +1,140 @@
+// Copyright 2016, EMC
+
+'use strict';
+
+var fs = require('fs');
+
+function TaskAnnotation(validator) {
+    this.validator = validator;
+}
+
+TaskAnnotation.prototype.mergeSchema = function (obj) {
```

Comment:
Could you please add some comment for each function to help other people to understand the design?